app/src/js/reducers/collections.js

```diff
@@ -48,105 +42,116 @@ export const initialState = {
   updated: {}
 };
 
-export default function reducer (state = initialState, action) {
-  state = Object.assign({}, state);
-  const { id, data } = action;
-
-  switch (action.type) {
-    case COLLECTION:
-      const colName = id.split('___');
-      const collection = data.results.find(function (element) {
-        return element.name === colName[0];
-      });
-      set(state, ['map', id, 'inflight'], false);
-      set(state, ['map', id, 'data'], assignDate(collection));
-      del(state, ['deleted', id]);
-      break;
-    case COLLECTION_INFLIGHT:
-      set(state, ['map', id, 'inflight'], true);
-      break;
-    case COLLECTION_ERROR:
-      set(state, ['map', id, 'inflight'], false);
-      set(state, ['map', id, 'error'], action.error);
-      break;
+export default createReducer(initialState, {
+  [COLLECTION]: (state, action) => {
+    const { id, data } = action;
+    const colName = id.split('___');
+    const collection = data.results.find(function (element) {
+      return element.name === colName[0];
+    });
```

Comment:
nit: While you're in here, perhaps you could import `deconstructCollectionId` from `api/lib/utils.js` and change these line to the following:

```
const { id, data } = action;
const { name } = deconstructCollectionId(id);
const collection = data.results.find((element) => element.name === name);
```

No biggie if you choose not to. Just thought it might be nice to remove the "hard-coded" separator.

Reply:
So @cumulus/api is only a dev dependency, not a runtime dependency.   But I will use the add that function to our format library.